### ISSUE TITLE
Support include_what_you_use running on CI

### ISSUE THREAD

**Issue by mithro** [verilog-to-routing/vtr-verilog-to-routing#624]
#### Proposed Behaviour

[`include-what-you-use` is a tool](https://include-what-you-use.org/) which makes sure that your C and C++ files include the correct set of headers. This makes it easier to refactor code as it doesn't stop working randomly because a include is missing.

#### Current Behaviour

You can accidentally forget to add an include but the code still works because something else includes it.

#### Context

This is another code quality improvement tool.

**Comment by kmurray** [comment]
I agree cleaning up extra includes etc is good.

Any thoughts on just doing this periodically vs running on CI? 

I know CI is going to be more consistent, but this does seem like a lower priority issue.

**Comment by mithro** [comment]
@kmurray Once it has been cleaned up, I think we should just run it as part of the C++ formatting check job.

**Comment by vaughnbetz** [comment]
This makes sense to me.  Tim, do you want to drive it?  Happy to assign it to you, and put it in the C++ formatting check.

**Comment by vaughnbetz** [comment]
This is a good idea ... @AmirhosseinPoolad discuss and see if you can take this one on.

**Comment by AlexandreSinger** [comment]
I also agree that this is a good idea. I have found that several files in VTR have header files that they include which they do not use. This is most problematic in header files. Some header files include other header files that they do not use which causes a cascade of includes. This ultimately leads to a problem when modifying a single header file ends up causing practically all files in VTR to be rebuilt which is not ideal.

Transitive includes (including a header file that you do not use, but you use something that it includes) also has similar problems but are not nearly as bad, especially if we have already cleaned up truly unused header files.

I have this tool installed locally in my editor and have found that many files in VTR have includes which are unused. I have tried removing them when I can, but they are very tricky to remove due to header files including unused header files which many of the CPP files rely on.
